src/ng2-smart-table/components/cell/edit-cell.component.ts

```diff
@@ -10,9 +10,31 @@ export class EditCellComponent {
 
   @Input() cell: Cell;
   @Input() inputClass: string = '';
+  @ViewChild('dynamicTarget', {read: ViewContainerRef}) dynamicTarget: any;
+  customComponent: any;
 
   @Output() public edited: EventEmitter<any> = new EventEmitter<any>();
 
+  constructor(private resolver: ComponentFactoryResolver){
+  }
+
+  ngOnChanges(arg) {
```

Comment:
Since methods like
```
  onEdited(event): boolean {
    this.edited.next(event);
    return false;
  }

  onStopEditing(): boolean {
    this.cell.getRow().isInEditing = false;
    return false;
  }
```
will be common I would create a common class and then the default _edit-cell_ will simply extend that class while the _custom-cell_ will extend that and implement some other methods.

Hope this makes sense, otherwise if you prefer to duplicate those methods in both components let me know, thanks.